custom-metrics-stackdriver-adapter/pkg/adapter/translator/query_builder.go

```diff
@@ -232,16 +331,31 @@ func (qb QueryBuilder) validate() error {
 		return apierr.NewInternalError(fmt.Errorf("QueryBuilder tries to build with translator value: nil"))
 	}
 
-	if qb.pods != nil && qb.podNames != nil {
-		return apierr.NewInternalError(fmt.Errorf("both pods and podNames are set for QueryBuilder, expected only one of them should be used"))
+	if !qb.nodes.isNodesContainerEmpty() {
+		// node metric
+		if !qb.nodes.isNodesContainerValid() {
+			return apierr.NewInternalError(fmt.Errorf("invalid nodes parameter is set to QueryBuilder"))
```

Comment:
These errors are a little implementation specific. I wonder if we can find errors that are more actionable to users.

Reply:
I thought about it, this case actually shouldn't happen, since all scenarios are covered by  `!qb.nodes.isNodesContainerEmpty()`